File: src/main/java/net/hectus/neobb/turn/default_game/flower/TOxeyeDaisy.java
package net.hectus.neobb.turn.default_game.flower;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.buff.ChancedBuff;
import net.hectus.neobb.player.NeoPlayer;
import org.bukkit.block.Block;

import java.util.List;

public class TOxeyeDaisy extends FlowerTurn {
    public TOxeyeDaisy(NeoPlayer player) { super(player); }
    public TOxeyeDaisy(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public int cost() {
        return 3;
    }

    @Override
    public List<Buff> buffs() {
        return List.of(new ChancedBuff(new Buff.ExtraTurn(), 25));
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/default_game/flower/TOxeyeDaisy.kt
package net.hectus.neobb.turn.default_game.flower

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.buff.ChancedBuff
import net.hectus.neobb.buff.ExtraTurn
import net.hectus.neobb.player.NeoPlayer
import org.bukkit.block.Block

class TOxeyeDaisy(data: Block?, cord: Cord?, player: NeoPlayer?) : FlowerTurn(data, cord, player) {
    override val cost: Int = 3

    override fun buffs(): List<Buff<*>> {
        return listOf(ChancedBuff(25.0, ExtraTurn()))
    }
}
